spectrocloud/resource_cluster_group.go

```diff
@@ -0,0 +1,305 @@
+package spectrocloud
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/terraform-provider-spectrocloud/pkg/client"
+)
+
+func resourceClusterGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceClusterGroupCreate,
+		ReadContext:   resourceClusterGroupRead,
+		UpdateContext: resourceClusterGroupUpdate,
+		DeleteContext: resourceClusterGroupDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		SchemaVersion: 2,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the cluster group",
+			},
+			"context": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "tenant",
```

Comment:
We usually set project as default context.

Reply:
In this case it will be tenant as it's the first context where we are introducing cluster group.